set.mm

```diff
@@ -199795,31 +199797,39 @@ implemented as a function (a set of ordered pairs)".  The current
   $}
 
   $( Define the structure component index extractor.  See theorem ~ ndxarg to
-     understand its purpose.  The restriction to ` NN ` allows ` ndx ` to exist
-     as a set, since ` _I ` is a proper class.  In principle, we could have
-     chosen ` CC ` or (if we revise all structure component definitions such as
-     ~ df-base ) another set such as the natural ordinal numbers ` _om `
-     ( ~ df-om ).  (Contributed by NM, 4-Sep-2011.) $)
+     understand its purpose.  The restriction to ` NN ` ensures that ` ndx ` is
+     a set.  The restriction to some set is necessary since ` _I ` is a proper
+     class.  In principle, we could have chosen ` CC ` or (if we revise all
+     structure component definitions such as ~ df-base ) another set such as
+     the set of finite ordinals ` _om ` ( ~ df-om ).  (Contributed by NM,
+     4-Sep-2011.) $)
   df-ndx $a |- ndx = ( _I |` NN ) $.
 
   ${
     $d x A $.
-    $( Define slot extractor for posets and related structures.  Note that the
-       function argument can be any set, although it is meaningful only if it
-       is a member of ` Poset ` ( ~ df-poset ) when used for posets or of
-       ` Grp ` ( ~ df-grp ) when used from groups.  (Contributed by Mario
-       Carneiro, 22-Sep-2015.) $)
+    $( Define the slot extractor for extensible structures.  The class
+       ` Slot A ` is a function whose argument can be any set, although it is
+       meaningful only if that set is a member of an extensible structure (such
+       as a partially ordered set ( ~ df-poset ) or a group ( ~ df-grp )).
+
+       Note that ` Slot A ` is "evaluation at ` A ` ".  The class ` Slot `
```

Comment:
```suggestion
       Note that ` Slot A ` is implemented as "evaluation at ` A ` ". That is,
       ` ( Slot A `` S ) ` is defined to be ` ( S `` A ) ` , where ` A ` will
       typically be a small natural number. Each extensible structure ` S ` is a function on
       defined on specific natural number "slots", and this function extracts the value at
       a particular slot.
       
       The special "structure" ` ndx ` , defined as the identity function, can be used to extract
       the number ` A ` from a slot, since ` ( Slot A `` ndx ) = A ` . This is typically used to refer
       to the number of a slot when defining structures without having to expose
       the detail of what that number is.
       
       The class ` Slot `
```